src/context.c

```diff
@@ -453,7 +453,11 @@ ly_ctx_new_yldata(const char *search_dir, const struct lyd_node *tree, int optio
     LY_CHECK_ARG_RET(NULL, tree, ctx, LY_EINVAL);
 
     /* create a new context */
-    LY_CHECK_GOTO(ret = ly_ctx_new(search_dir, options, &ctx_new), cleanup);
+    if (*ctx == NULL) {
```

Comment:
API function functionality changed, please update its documentation accordingly.

Reply:
I also updated the documentation for ly_ctx_new_ylmem and ly_ctx_new_ylpath, since these call ly_ctx_new_yldata internally and are affected by the change.